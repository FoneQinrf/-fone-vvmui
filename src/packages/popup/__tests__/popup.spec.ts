/*
 * @Author: Fone`峰
 * @Date: 2021-05-16 12:10:17
 * @LastEditors: Fone`峰
 * @LastEditTime: 2021-05-16 12:36:17
 * @Description: file content
 * @Email: qinrifeng@163.com
 * @Github: https://github.com/FoneQinrf
 */
import { mount } from '@vue/test-utils';
import Popup from '../index.vue';
import { nextTick, createVNode } from 'vue';

describe('Popup', () => {
  const wrapper = mount(Popup, { props: { isTeleport: true } });

  it('should render position corretly', async () => {
    wrapper.setProps({
      show: true
    });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.setProps({ position: 'top' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.setProps({ position: 'bottom' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.setProps({ position: 'right' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.setProps({ position: 'left' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should click overlay corretly', async () => {
    wrapper.find('.vvm-popup--overlay').trigger('click');
    await nextTick();
    expect(wrapper.emitted()).toHaveProperty('update:show');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render overlay corretly', async () => {
    wrapper.setProps({ overlay: false });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.vvm-popup--overlay').exists()).toBe(false);
  });

  it('should render slot corretly', async () => {
    const popup = mount(Popup, {
      slots: {
        default: () => createVNode('span', {}, 'render slot')
      },
      props: {
        show: true,
        isTeleport: true
      }
    });
    await nextTick();
    expect(popup.html()).toMatchSnapshot();
    expect(popup.find('.vvm-popup').find('span').exists()).toBe(true);
  });
});
