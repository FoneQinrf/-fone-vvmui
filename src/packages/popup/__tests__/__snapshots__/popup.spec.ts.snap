// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should click overlay corretly 1`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"vvm-popup--overlay\\"></div>
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--left\\"></div>
</transition-stub>
<!--teleport end-->"
`;

exports[`Popup should render overlay corretly 1`] = `
"<!--teleport start-->
<transition-stub>
  <!--v-if-->
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--left\\"></div>
</transition-stub>
<!--teleport end-->"
`;

exports[`Popup should render position corretly 1`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"vvm-popup--overlay\\"></div>
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--center\\"></div>
</transition-stub>
<!--teleport end-->"
`;

exports[`Popup should render position corretly 2`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"vvm-popup--overlay\\"></div>
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--top\\"></div>
</transition-stub>
<!--teleport end-->"
`;

exports[`Popup should render position corretly 3`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"vvm-popup--overlay\\"></div>
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--bottom\\"></div>
</transition-stub>
<!--teleport end-->"
`;

exports[`Popup should render position corretly 4`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"vvm-popup--overlay\\"></div>
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--right\\"></div>
</transition-stub>
<!--teleport end-->"
`;

exports[`Popup should render position corretly 5`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"vvm-popup--overlay\\"></div>
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--left\\"></div>
</transition-stub>
<!--teleport end-->"
`;

exports[`Popup should render slot corretly 1`] = `
"<!--teleport start-->
<transition-stub>
  <div class=\\"vvm-popup--overlay\\"></div>
</transition-stub>
<transition-stub>
  <div class=\\"vvm-popup vvm-popup--center\\"><span>render slot</span></div>
</transition-stub>
<!--teleport end-->"
`;
